packages/user-extension/src/index.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module user-extension
+ */
+
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { PageConfig } from '@jupyterlab/coreutils';
+import { IStateDB } from '@jupyterlab/statedb';
+import {
+  ICurrentUser,
+  IMenu,
+  IUser,
+  IUserMenu,
+  RendererUserMenu,
+  User,
+  UserMenu
+} from '@jupyterlab/user';
+import { MenuBar, Widget } from '@lumino/widgets';
+
+/**
+ * Jupyter plugin providing the ICurrentUser.
+ */
+const userPlugin: JupyterFrontEndPlugin<IUser> = {
+  id: '@jupyterlab/user-extension:user',
+  autoStart: true,
+  requires: [IStateDB],
+  provides: ICurrentUser,
+  activate: (app: JupyterFrontEnd, state: IStateDB): IUser => {
+    return new User(state);
+  }
+};
+
+/**
+ * Jupyter plugin providing the IUserMenu.
+ */
+const userMenuPlugin: JupyterFrontEndPlugin<IMenu | undefined> = {
+  id: '@jupyterlab/user-extension:userMenu',
+  autoStart: true,
+  requires: [ICurrentUser],
+  provides: IUserMenu,
+  activate: (app: JupyterFrontEnd, user: IUser): IMenu | undefined => {
+    const { shell, commands } = app;
+
+    if (PageConfig.getOption('collaborative') !== 'true') {
+      return undefined;
+    }
+
+    const spacer = new Widget();
+    spacer.id = 'jp-topbar-spacer';
+    shell.add(spacer, 'top', { rank: 900 });
```

Comment:
It's fine to have the spacer here for now, but maybe we should check how to make the top area more well defined so extensions can more easily add items there without stepping on each others toes.

For example [jupyterlab-topbar](https://github.com/jupyterlab-contrib/jupyterlab-topbar) also adds a spacer to align the items to the right. Which means there would be two spacers if the user menu and the top bar extensions (or any other extensions) are activated on the page.

So maybe we should just add that spacer independently of these plugins, by default in JupyterLab, and document using a rank < 900 to place items to the left, and rank > 900 to place them to the right.

Reply:
Opened https://github.com/jupyterlab/jupyterlab/issues/11589

Reply:
Yep, good point.